dm/worker/metrics.go

```diff
@@ -34,14 +34,27 @@ import (
 	"github.com/pingcap/dm/syncer"
 )
 
+const (
+	opErrTypeBeforeOp    = "BeforeAnyOp"
+	opErrTypeSourceBound = "SourceBound"
+)
+
 var (
 	taskState = metricsproxy.NewGaugeVec(
 		prometheus.GaugeOpts{
 			Namespace: "dm",
 			Subsystem: "worker",
 			Name:      "task_state",
 			Help:      "state of task, 0 - invalidStage, 1 - New, 2 - Running, 3 - Paused, 4 - Stopped, 5 - Finished",
-		}, []string{"task"})
+		}, []string{"task", "source_id"})
+
+	opErrCounter = metricsproxy.NewCounterVec(
```

Comment:
how to use this metric? send an alert message when the value is changed?

Reply:
show in grafana for example, user could thus know worker's state and decide to retire this worker or add more worker.